### DISCUSSION TITLE
One Graph, and inputs are features

### DISCUSSION THREAD

**Discussion by hossam-zaki** [pyg-team/pytorch_geometric#2767] [Q&A]
Hello! I'm learning a bit more about Pytorch geometric for a project I'm working on. My original plan was to have one graph, and the inputs would be the features for each node in the graph. But, from what I'm reading, this doesn't seem to be possible. Is it?

Another alternative would be to have the standard inputs (graphs and features in the Data object), where the adjacency matrix will not change, but the features will. Would this work too?

Any help is appreciated. Thank you so much!

**Comment by rusty1s** [comment]
Thanks for your interest. I'm not entirely sure what you are trying to do. PyG can handle both processing of a single graph (e.g., `examples/gcn.py`) or multiple graphs (e.g., `examples/mutag_gin.py`). If your adjacency matrix is fixed and only your input feature matrix vary, I think it's best to assign your `edge_index` to all graphs and store them in a list of `data` objects.

**Reply by hossam-zaki** [reply]
Thanks a lot for the comment. Yes, the adjacency matrix is fixed, and the input feature matrix will vary. So basically, you are suggesting that I process multiple graphs, each with the same adjacency matrix but different input features?

**Reply by rusty1s** [reply]
Yes, as long as you take care of sharing memory of `edge_index` between different `data` objects, there shouldn't be any memory overhead on the CPU side. On the GPU, `edge_index` will be replicated for each example in your batch, which is probably a negligible overhead.
